### DISCUSSION TITLE
[Question]: Can Carbon work with Remix

### DISCUSSION THREAD

**Discussion by tlawrie** [carbon-design-system/carbon#11775] [Help] [answered]
### Question for Carbon

We are currently doing initial feasability PoC's with Remix.run and want to understand if there are documented install instructions and tips for using Remix with Carbon's React components?

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by timrbula** [comment, marked as answer]
Hi @tlawrie. I'm not aware of any official guides for Remix, but I was able to get it working.  Take a look at [this project in StackBlitz](https://stackblitz.com/edit/remix-carbon?file=app/root.tsx).

I needed to add compilation for the styles via the `sass` script and then reference the css file in the `root.tsx` file. I also had to include Plex via Google Fonts (requests for them were returning a 404 that I didn't look into). Overall, not too bad at all to get things going. Let me know if you have any questions on the above. 

It would be great to have someone from the Carbon team to chime on anything more official though.

**Reply by joshblack** [reply]
Thanks for this @timrbula! Really appreciate it 🙏 We'll see if we can move an example over into the project based on this to help out others in the future 👍

**Comment by tlawrie** [comment]
Thanks @timrbula that is super helpful!

With the SASS scripts, I see that it relates to [Remix.run Preprocessor Styling](https://remix.run/docs/en/v1/guides/styling#css-preprocessors) do you know if the approach being taken is then using [shared component styles](https://remix.run/docs/en/v1/guides/styling#shared-component-styles)?

**Comment by timrbula** [comment]
@tlawrie Yes that's right. I am loading all of the carbon styles in the `root.tsx` of the application so all routes have access to them. Then you can add additional styles per route that get dynamically added/removed as the route is accessed.

**Comment by tlawrie** [comment]
hey @joshblack do you want an issue for creating / tracking a reference template for Remix + Carbon? A fully fleshed out working example.

**Comment by vijayganeshpk** [comment]
Thank you @timrbula , based on your example I got my version working with Remix & Vite.

Please check the [sample repo](https://github.com/vijayganeshpk/web-ui-libraries-eval/tree/master/with-remix/ibm-carbon) for reference.

It is downloading a 3mb JS at the start, need to find optimization for the same.
